src/ui/uas/UASMessageView.cc

```diff
@@ -0,0 +1,159 @@
+}
+
+/*-------------------------------------------------------------------------------------
+  UASMessageViewWidget
+-------------------------------------------------------------------------------------*/
+
+UASMessageViewWidget::UASMessageViewWidget(QWidget *parent)
+    : UASMessageView(parent)
+    , _unconnectedWidget(NULL)
+{
+    setStyleSheet("QPlainTextEdit { border: 0px }");
+    // Construct initial widget
+    _unconnectedWidget = new QGCUnconnectedInfoWidget(this);
+    ui()->horizontalLayout->addWidget(_unconnectedWidget);
+    ui()->plainTextEdit->hide();
+    // Enable the right-click menu for the text editor. This works because the plainTextEdit
+    // widget has its context menu policy set to its actions list. So any actions we add
+    // to this widget's action list will be automatically displayed.
+    // We only have the clear action right now.
+    QAction* clearAction = new QAction(tr("Clear Text"), this);
+    connect(clearAction, SIGNAL(triggered()), ui()->plainTextEdit, SLOT(clear()));
+    ui()->plainTextEdit->addAction(clearAction);
+    // Connect message handler
+    connect(UASMessageHandler::instance(), SIGNAL(textMessageReceived(UASMessage*)), this, SLOT(handleTextMessage(UASMessage*)));
+}
+
+UASMessageViewWidget::~UASMessageViewWidget()
+{
+
+}
+
+void UASMessageViewWidget::handleTextMessage(UASMessage *message)
+{
+    // Reset
+    if(!message) {
+        ui()->plainTextEdit->clear();
+        _unconnectedWidget->show();
+        ui()->plainTextEdit->hide();
+    } else {
+        // Make sure the UI is configured for showing messages.
+        // Note that this call is NOT equivalent to `_unconnectedWidget->isVisible()`.
+        if (!_unconnectedWidget->isHidden())
+        {
+            _unconnectedWidget->hide();
+            ui()->plainTextEdit->show();
+        }
+        QPlainTextEdit *msgWidget = ui()->plainTextEdit;
+        // Turn off updates while we're appending content to avoid breaking the autoscroll behavior
+        msgWidget->setUpdatesEnabled(false);
+        QScrollBar *scroller = msgWidget->verticalScrollBar();
+        msgWidget->appendHtml(message->getFormatedText());
+        // Ensure text area scrolls correctly
+        scroller->setValue(scroller->maximum());
+        msgWidget->setUpdatesEnabled(true);
+    }
+}
+
+/*-------------------------------------------------------------------------------------
+  UASMessageViewRollDown
+-------------------------------------------------------------------------------------*/
+
+UASMessageViewRollDown::UASMessageViewRollDown(QWidget *parent, QGCToolBar *toolBar)
+    : UASMessageView(parent)
+{
+    _toolBar = toolBar;
+    setAttribute(Qt::WA_TranslucentBackground);
+    setStyleSheet("background-color: rgba(0%,0%,0%,80%); border: 2px;");
+    QPlainTextEdit *msgWidget = ui()->plainTextEdit;
+    QAction* clearAction = new QAction(tr("Clear Text"), this);
+    connect(clearAction, SIGNAL(triggered()), msgWidget, SLOT(clear()));
```

Comment:
Probably because the signal connection is broken. Look in your debug output. I always run with QT_FATAL_WARNINGS=1 in my config. This turns all qWarnings into asserts which will stop the run.